sources/api/early-boot-config/Cargo.toml

```diff
@@ -18,13 +18,15 @@ log = "0.4"
 reqwest = { version = "0.10", default-features = false, features = ["blocking"] }
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1"
+serde_plain = "0.3"
 serde-xml-rs = "0.4.1"
 simplelog = "0.9"
 snafu = "0.6"
 tokio = { version = "0.2", default-features = false, features = ["macros", "rt-threaded"] }
 # When hyper updates to tokio 0.3:
 #tokio = { version = "0.3", default-features = false, features = ["macros", "rt-multi-thread"] }
 toml = "0.5"
+vmw_backdoor = "0.2"
```

Comment:
vmw_backdoor only works on x86_64 (using x86 asm) and I believe VMware only supports x86 virtualization (though it can now run on ARM hosts), so we might want to do something to stop BUILDSYS_ARCH=aarch64 BUILDSYS_VARIANT=vmware-dev.  Maybe we should check the arch/variant combo in Makefile.toml so we don't get this far; it seems like any solution localized to early-boot-config could raise issues regarding whether conditional compilation / arch-specific errors would break unit tests, etc.  And maybe just a comment here?

Reply:
Good point - I think a solution in `Makefile.toml` is the way to go.

Reply:
I'll probably have to create a platform specific dependency for `vmw_backdoor` in `Cargo.toml` anyway, just to keep `early-boot-config` from failing to build for ARM.

Reply:
> Good point - I think a solution in Makefile.toml is the way to go.

Makefile solutions mean that we can't use our workspace for development anymore :(

Reply:
P.S. variants that only work for some arches seems like a new concept.

Reply:
> Makefile solutions mean that we can't use our workspace for development anymore

What does this mean?  What workspace?  What would break?

> variants that only work for some arches seems like a new concept.

It is, one that we have to tackle somehow until VMware supports ARM guests...

> I'll probably have to create a platform specific dependency for vmw_backdoor in Cargo.toml anyway, just to keep early-boot-config from failing to build for ARM.

Having an early-boot-config on VMware that doesn't support guestinfo would be kind of unhelpful, right?  Seems like we should prevent that from existing at all, with a higher level solution..?

Reply:
I think we're going to have to do some combination of both a `Makefile` solution and a crate level solution at the `early-boot-config` level.  If you look at CI (and I'm able to reproduce this locally), _every_ variant fails to build for ARM because `vmw_backdoor` is a dependency of `early-boot-config` and includes x86 assembly.  Even though `vmw_backdoor` crate [compiles to an empty crate if not on x86](https://github.com/lucab/vmw_backdoor-rs/blob/ff94a690bf813a1e1915ff774a54ea88fe9b6f51/src/lib.rs#L51) it's still getting compiled.  See below error messages from the build.

```
   #17 11.78 warning: src/asm/x86_64-linux.s:152: Error: unknown mnemonic `movq' -- `movq %r9,%rbp'
  #17 11.78 warning: src/asm/x86_64-linux.s:155: Error: unknown mnemonic `xor' -- `xor %rax,%rax'
  #17 11.78 
  #17 11.78 error: failed to run custom build command for `vmw_backdoor v0.2.0`
```

My suggested solution:
* In `Makefile.toml`, prevent users from building the `vmware*` / `aarch64` combo.
* In `early-boot-config/Cargo.toml`, prevent the `vmw_backdoor` crate from being build for `aarch64` with the addition of below.
```
+[target.'cfg(target_arch = "x86_64")'.dependencies]                                                                                              
 vmw_backdoor = "0.2"
```

Reply:
> What does this mean? What workspace? What would break?

If I understand the problem correctly, this would not work on an ARM developer machine:

```sh
cd sources && export VARIANT=anyvariant? && cargo build
```

What's not clear in my mind is the list of VARIANT values that would fail to build (only VMWARE? in that case maybe its fine since it would be reasonable to assume you can't compile VMWare bits on an arm machine in that way.

Reply:
Does this help? https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#platform-specific-dependencies

Reply:
Sorry, Zack already found the answer before my previous two comments:

> [target.'cfg(target_arch = "x86_64")'.dependencies]

Reply:
@bcressey made a good point: Another (probably more preferable) solution would be to specify the supported architectures for a variant in the variant's `Cargo.toml`.

Reply:
Opened #1404 to track this feature for `buildsys`